test/DynamicDependency/Test_Win32/Test_GetCurrentPackageInfo.cpp

```diff
@@ -360,17 +382,42 @@ namespace Test::DynamicDependency
                     }
                 }
 
-                m_kernelbaseDll = std::move(dll);
                 m_getCurrentPackageInfo3 = std::move(function);
             }
         }
 
         bool IsGetCurrentPackageInfo3Supported()
         {
-            EnsureGetCurrentPackageInfo3IsIniitalizedIfExists();
+            EnsureGetCurrentPackageInfo3IsInitializedIfExists();
             return m_getCurrentPackageInfo3 != nullptr;
         }
 
+        void EnsureGetPackageGraphRevisionIdIsInitializedIfExists()
+        {
+            EnsureKernelbaseDllIsLoaded();
+
+            if (!m_getPackageGraphRevisionId)
+            {
+                auto function{ reinterpret_cast<GetPackageGraphRevisionIdFunction>(GetProcAddress(m_kernelbaseDll.get(), "GetPackageGraphRevisionId")) };
+                if (!function)
+                {
+                    const auto lastError{ GetLastError() };
+                    if (lastError != ERROR_PROC_NOT_FOUND)
+                    {
+                        FAIL_FAST_WIN32(lastError);
+                    }
+                }
+
+                m_getPackageGraphRevisionId = std::move(function);
+            }
+        }
+
+        bool IsGetPackageGraphRevisionIdSupported()
+        {
+            EnsureGetPackageGraphRevisionIdIsInitializedIfExists();
```

Comment:
Shouldn't this be called only when m_getPackageGraphRevisionId == nullptr to avoid re-running same code in every call  ?

Reply:
`EnsureGetPackageGraphRevisionIdIsInitializedIfExists()` does the checks to only do work once and cache process-constant info for latest calls. Keeps the is-it-available-or-is-it-not isolated low and not pollute higher level checks, and the work to hit this 2nd+ time is cheap and this is a test not product code so I favored maintainability over ultra squeezing perf